In [1]:
import requests as re
from geo_extractor import get_bellingcat_data, get_ceninfores_data, get_defmon_data\
    , get_defmon_spreadsheet_data, get_geoconfirmed_data, get_texty_data, extractors, TextyExtractor
import pandas as pd
import json

In [27]:
db_list_func = [get_bellingcat_data(), get_ceninfores_data(), get_defmon_data(), 
                get_defmon_spreadsheet_data(), get_geoconfirmed_data(), get_texty_data()
               ]

In [28]:
db_str_func = ['get_bellingcat_data', 'get_ceninfores_data', 'get_defmon_data', 
                'get_defmon_spreadsheet_data', 'get_geoconfirmed_data', 'get_texty_data'
               ]

In [29]:
#size of dbs
for i in db_list_func:
    events_df = pd.DataFrame(i)
    print(f"size {events_df.shape}")

size (1115, 10)
size (10006, 10)
size (0, 0)
size (19272, 10)
size (9047, 10)
size (25884, 10)


Shape of dbs in above list function - April 3, 2023
- size (1079, 9)
- size (9608, 9)
- size (0, 0)
- size (19272, 9)
- size (8110, 9)
- size (24398, 9)

Shape of dbs in above list function - April 4, 2023
- size (1083, 9)
- size (9608, 9)
- size (0, 0)
- size (19272, 9)
- size (8119, 9)
- size (24487, 9)

Shape of dbs in above list function - April 17, 2023
- size (1115, 9)
- size (10006, 9)
- size (0, 0)
- size (19272, 9)
- size (8876, 9)
- size (25434, 9)

Shape of dbs in above list function - April 17, 2023
- size (1115, 10)
- size (10006, 10)
- size (0, 0)
- size (19272, 10)
- size (9047, 10)
- size (25884, 10)

In [30]:
events = get_bellingcat_data()
events_df = pd.DataFrame(events)
# for e in events[:10]:
#     print(f"{e.id}: [{e.latitude}, {e.longitude}], {e.description}")

In [6]:
events_df.head()

id       date   latitude  longitude                 place_desc title  \
0  CIV0001 2022-02-24  49.850050  36.659031  Chuhuiv, south of Kharkiv  None   
1  CIV0002 2022-02-24  48.748564  30.218515                       Uman  None   
2  CIV0003 2022-02-24  50.470055  30.527381                       Kyiv  None   
3  CIV0004 2022-02-24  47.775537  37.239601                   Vuhledar  None   
4  CIV0007 2022-02-24  46.227890  34.642830        Semihatka/Henichesk  None   

                                         description      source  \
0  Apartment block hit. Crater is very large, pos...  BELLINGCAT   
1  Civilians hit by what appears to have been art...  BELLINGCAT   
2    Explosion in central Kyiv, nothing further yet.  BELLINGCAT   
3    Explosion in central Kyiv, nothing further yet.  BELLINGCAT   
4  [Graphic: Image shows a dead child wrapped in ...  BELLINGCAT   

                                               links  \
0  [https://twitter.com/Michael1Sheldon/status/14...   
1  [https://twitter.com/ragipsoylu/status/1496737...   
2  [https://twitter.com/FannyFacsar/status/149678...   
3  [https://twitter.com/IBN_MOHAMMAD/status/14967...   
4  [https://twitter.com/GayanaYuksel/status/14968...   

                                             filters  
0  [{'key': 'Type of area affected', 'value': 'Re...  
1  [{'key': 'Type of area affected', 'value': 'Co...  
2  [{'key': 'Type of area affected', 'value': 'Un...  
3  [{'key': 'Type of area affected', 'value': 'He...  
4  [{'key': 'Type of area affected', 'value': 'Un...

What I discovered by just downloading the entire JSON is that there is a filter dictionary within each entry that contains a list of sources and the type of activity. I need to see if I can pull that JSON into the notebook. This package function only extracts the JSON of data for the entry in the DB, and misses the filter information.

In [8]:
events_df.shape

(1115, 9)

In [9]:
data = events[:40]

In [15]:
type(data)

list

In [31]:
events_df.to_csv('bellingcat.csv', index = False)

In [7]:
events = get_texty_data()
events_df = pd.DataFrame(events)
events_df.to_csv('texty.csv', index = False)

In [8]:
events_df.tail()

id       date   latitude  longitude  \
25879  None 2023-04-19  46.642848  32.631768   
25880  None 2023-04-19  46.865619  33.575855   
25881  None 2023-04-19  46.750323  33.017671   
25882  None 2023-04-19  46.706394  32.690588   
25883  None 2023-04-19  46.590002  32.465159   

                                              place_desc         title  \
25879  Будівля підприємства критичної інфраструктури ...        Херсон   
25880                                         Херсонська       Зміївка   
25881                                         Херсонська     Іванівка    
25882                                         Херсонська    Інженерне    
25883                                         Херсонська  Дніпровське    

      description source                                              links  \
25879        None  TEXTY  [https://www.pravda.com.ua/news/2023/04/19/739...   
25880        None  TEXTY  [https://www.pravda.com.ua/news/2023/04/20/739...   
25881        None  TEXTY  [https://www.pravda.com.ua/news/2023/04/20/739...   
25882        None  TEXTY  [https://www.pravda.com.ua/news/2023/04/20/739...   
25883        None  TEXTY  [https://www.pravda.com.ua/news/2023/04/20/739...   

      filters  
25879      []  
25880      []  
25881      []  
25882      []  
25883      []

In [10]:
#texty data is in cyrillic, so it's unreadable. In addition, the sources are largely from pravda, which is a Ukrainian state publication
#maybe I can look at creating a translator, but I am also unsure if I want to use this source for the capstone

@Geoconfirmed has an api to directly gather all displayed data from the web map in a JSON format:
https://geoconfirmed.azurewebsites.net/api/placemark/Ukraine

It has a classification engine in a key called "icon", which determines what image is placed on the map. This seems to be the way the data is labeled, and it has an actor and an ID contained within this label. I could either
* use this icon to train a classification model (because this would be the Y value in the data). **or**
* use the icon as a variable to train a classification model on new classes 

I think I am going to go with the second option. There are a lot of different types of classifications to have to parse through from the application, and including it as a variable in the classification model might be more beneficial.

In [5]:
res = re.get('https://geoconfirmed.azurewebsites.net/api/placemark/Ukraine')

In [6]:
res.status_code

200

In [8]:
geoconfirmed_json = res.json()

In [12]:
type(geoconfirmed_json) #oh, it is a list with nested dictionaries, like in the SQL Hackathon

list

In [20]:
geoconfirmed_json[200]

{'id': '2a7346f5-520e-4b43-1414-08dabc42150c',
 'date': '2022-02-26T13:00:00',
 'name': '26 1300 FEB 2022 - PIC - Destroyed Russian Armor in #Monachynivka, 26 FEB 2022',
 'description': 'https://twitter.com/naalsio26/status/1516967470577373185 <br><br>Geo: <br>https://twitter.com/GeoConfirmed/status/1517627835724218368',
 'coordinates': [49.819446, 37.5335],
 'icon': '_CIRCLES_E00000_RU_56_42.png',
 'originalSource': 'https://twitter.com/naalsio26/status/1516967470577373185',
 'geolocation': 'https://twitter.com/GeoConfirmed/status/1517627835724218368',
 'origin': 1,
 'allGeolocations': False}

In [21]:
"""I can do what I did in the SQL Hackathon to break this into a series of lists that can be columnized into a pandas dataframe and exported
The data are not as neatly organized as they are from the geo extractor, but, the geoextractor is missing the icon column
I can join the icon column to the main geoconfirmed dataframe so that way it is included
"""
geoconfirmed_id = []
icon = []

for i in geoconfirmed_json:    
    geoconfirmed_id.append(i['id'])
    icon.append(i['icon'])

In [25]:
geoconfirmed_df = pd.DataFrame({'geoconfirmed_id': geoconfirmed_id, 'icon': icon})

In [32]:
events = get_geoconfirmed_data()

In [33]:
events_df = pd.DataFrame(events)

In [36]:
#merge geoconfirmed_df and events_df
merged_geo_df = geoconfirmed_df.merge(events_df, left_on = 'geoconfirmed_id', right_on = 'id')

In [37]:
#dataframe and export
merged_geo_df.to_csv('get_geoconfirmed_data.csv', index = False)

In [38]:
events = get_ceninfores_data()
events_df = pd.DataFrame(events)
events_df.to_csv('ceninfores.csv', index = False)

In [39]:
events = get_defmon_spreadsheet_data()
events_df = pd.DataFrame(events)
events_df.to_csv('defmon.csv', index = False)